stacker/config/constructors.py

```diff
@@ -0,0 +1,39 @@
+import json
+import subprocess
+import yaml
+
+
+def get_vaulted_value(value):
+    """Get the specified value from vault.
+
+    Vaulted field types should use the following format:
+
+        <path within vault>@<key within stored vault data>
+
+    For example:
+
+        # We've set the secret/hello key with the following
+        $ vault write secret/hello value=world
+
+        # In stacker we would reference "world" with the following
+        conf_key: !vault secret/hello@value
+
+        # The above would resolve to
+        conf_key: world
+
+    """
+    if '@' not in value:
+        raise TypeError('Vaulted vaules must include a "@<key in stored data>"')
+
+    path, key = value.split('@')
+    response = subprocess.check_output(['vault', 'read', '-format=json', path])
```

Comment:
they do have an HTTP API, but then I have to know more about the vault endpoint that is set as well as auth. if i was going to require having that setup knowledge, i would just want to think about how we let other people register these and then be able to use subprocess.

what if we had a "translators" parameter?

```
stacker build -r us-east-1 conf/staging.env conf/example.yaml --translators my_private_repo/translators.py
```